sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -796,3 +797,71 @@ case class SchemaOfJson(
 
   override def prettyName: String = "schema_of_json"
 }
+
+/**
+ * A function which returns all the keys of outer JSON object.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(json_object) - returns all the keys of outer JSON object.",
+  arguments = """
+    Arguments:
+      * json_object - A JSON object. If it is an invalid string, the function returns null.
+          If it is a JSON array, an illegal argument exception will be thrown.
+  """,
+  examples = """
+    Examples:
+      > Select _FUNC_('{}');
+        []
+      > Select _FUNC_('{"key": "value"}');
+        ["key"]
+      > Select _FUNC_('{"f1":"abc","f2":{"f3":"a", "f4":"b"}}');
+        ["f1","f2"]
+  """,
+  since = "3.1.0")
+case class JsonObjectKeys(child: Expression) extends UnaryExpression with CodegenFallback
+  with ExpectsInputTypes {
+
+  override def inputTypes: Seq[DataType] = Seq(StringType)
+  override def dataType: DataType = ArrayType(StringType)
+  override def nullable: Boolean = true
+  override def prettyName: String = "json_object_keys"
+
+  override def eval(input: InternalRow): Any = {
+    val json = child.eval(input).asInstanceOf[UTF8String]
+    if(json == null) {
+      return null
+    }
+
+    try {
+      Utils.tryWithResource(CreateJacksonParser.utf8String(SharedFactory.jsonFactory, json)) {
+        parser => getJsonKeys(parser, input)
+      }
+    } catch {
+      case _: JsonProcessingException => null
+    }
+  }
+
+  private def getJsonKeys(parser: JsonParser, input: InternalRow): Any = {
+    var arrayBufferOfKeys = ArrayBuffer.empty[UTF8String]
+    // this handles `NULL` case
```

Comment:
I understand that you meant `parser.nextToken() == null`, but shall we update the comment like `This handles empty string case`?